listmonk@.service

```diff
@@ -0,0 +1,69 @@
+[Unit]
+Description=listmonk mailing list and newsletter manager (%I)
+ConditionPathExists=/etc/listmonk/%i.toml
+Wants=network.target
+# The PostgreSQL database may not be on the same host but if it
+# is listmonk should wait for it to start up.
+After=postgresql.service
+
+[Service]
+Type=simple
+EnvironmentFile=-/etc/default/listmonk
+EnvironmentFile=-/etc/default/listmonk-%i
+ExecStartPre=/usr/bin/mkdir -p "${HOME}/uploads"
+ExecStartPre=/usr/bin/listmonk --config /etc/listmonk/%i.toml --upgrade --yes
```

Comment:
DB upgrade, while ideally safe an idempotent, is still a "destructive" action that should require explicit user consent, given the end user an opportunity to backup the DB. Not sure if it should implicitly executed.

Reply:
Personally I’d value the continuous operation of the service higher than the little feel of control that comes with the manual execution of database migrations. Either I’m performing automated backups, in which case I would want the service to automatically execute migrations because I can always rollback or I did not configure backups and I’m taking the lets-hope-for-the-best road anyway :). But I understand that this is a debatable opinion :).

I see a few options:
1. no automatic migrations at all: In that case it would be nice if listmonk could go into a failed state if the expected migration version does not match the database version. The question is if this failed state should be visible to the system administrator or the listmonk administrator. I vote it should be the system administrator because they’re the only one who can fix it, which means that the service should exit with a non-zero exit code if migrations haven’t been executed.
2. backup before migrate: listmonk automatically performs a backup with the configured credentials before executing migrations. Possibly enabled through a cli option like `--backup`.
3. configuration-level consent: only automatically execute migrations if consent has been given through configuration (either an env variable, a configuration option or even both).
4. always migrate but advise backups: advise users to configure backups in the project/systemd unit documentation and suggest appropriate solutions with os-specific packaging like [`autopostgresqlbackup`](https://packages.debian.org/stretch/admin/autopostgresqlbackup).

Reply:
1 is already the current behaviour. If the binary changes and if it's missing a required DB migration, it exits with a non-zero code and prints a message on stdout prompting the user to run `--upgrade`.

Thinking about it, we can go with the PR (`ExecStartPre`) and advise users to do a DB backup before upgrading the binary by documenting it in the unit file. Eventually, a `--backup` shortcut can be added to make backups easy.